### PR TITLE
[windows] Added my_timezone and my_daylight functions for win32.

### PR DESCRIPTION
**Pull request by de-wim**
This fixes compilation using VS2015 on my machine, though linking does not work yet.

### CODE REVIEW

**Comment by de-wim** [comment]
The second commit fixes linking on my system.

File: to-be-migrated/time/time-portability.h [hunk 1]
```diff
@@ -2,6 +2,10 @@
 #define __extension__
 #include <machine/types.h>
 #endif
+#if defined(_WIN32)
+#include "Windows.h"
+#endif
+
```
Comment (waywardmonkeys) [inline to-be-migrated/time/time-portability.h]: This doesn't need to be in the header, does it?

File: to-be-migrated/time/time-portability.c [hunk 2]
```diff
@@ -41,6 +41,29 @@ my_daylight(void)
 {
   return _daylight;
 }
+#elif defined(_WIN32)
+int
+my_timezone(void)
+{
+  DYNAMIC_TIME_ZONE_INFORMATION tzinfo;
+  DWORD rv = GetDynamicTimeZoneInformation(&tzinfo);
+  if(rv == TIME_ZONE_ID_UNKNOWN) // No zone info, just return 0
```
Comment (waywardmonkeys) [inline to-be-migrated/time/time-portability.c]: Missing space between `if` and `(`.

**Comment by waywardmonkeys** [comment]
A couple of requests:

1) Can you pull the 2 linking fixes into 2 separate pull requests or at least not put them here? I think we want to do something different in both cases.

2) The `CMakeLists.txt` shouldn't do the checks for time / daylight stuff on Windows now that we have a proper Windows implementation.

File: to-be-migrated/time/time-portability.h [hunk 3]
```diff
@@ -2,6 +2,7 @@
 #define __extension__
 #include <machine/types.h>
 #endif
+
```
Comment (waywardmonkeys) [inline to-be-migrated/time/time-portability.h]: This isn't needed.

File: to-be-migrated/time/time-portability.c [hunk 4]
```diff
@@ -41,6 +41,31 @@ my_daylight(void)
 {
   return _daylight;
 }
+#elif defined(_WIN32)
+#include "Windows.h"
```
Comment (waywardmonkeys) [inline to-be-migrated/time/time-portability.c]: This should be `<Windows.h>` and not `"Windows.h"`.

**Pull request merged**